lib/elixir/test/elixir/task_test.exs

```diff
@@ -493,6 +493,7 @@ defmodule TaskTest do
     test "timeout" do
       assert catch_exit([:infinity] |> Task.async_stream(&sleep/1, [timeout: 0]) |> Enum.to_list) ==
              {:timeout, {Task.Supervised, :stream, [0]}}
+      refute_receive _
```

Comment:
Sorry I should have explained this. I changed it because `refute_received` did not catch this bug before because the message would be sent after the exit being caught and the `refute_received`. It actually passes from time to time even with `refute_receive` so I am not entirely sure how to proceed. 😕